src/coreclr/src/jit/hwintrinsicarm64.cpp

```diff
@@ -281,7 +281,9 @@ GenTree* Compiler::impSpecialIntrinsic(NamedIntrinsic        intrinsic,
 
         if (!varTypeIsArithmetic(baseType))
         {
-            assert((intrinsic == NI_Vector64_AsByte) || (intrinsic == NI_Vector128_As));
+            assert((intrinsic == NI_Vector64_AsByte) || (intrinsic == NI_Vector128_As) ||
```

Comment:
@echesakovMSFT, not related to this PR, but I think the first check in this is wrong. It should be `intrinsic == NI_Vector64_As`, right?

Reply:
It looks so, however, I don't see Vector64.As intrinsic in hwintrinsiclistarm64.h, I will update #33308 to include this.